src/general/FilterArgument.cs

```diff
@@ -0,0 +1,59 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+
+/// <summary>
+///   Parent class for filters arguments
+/// </summary>
+/// <remarks>
+///   <para>
+///     Used to list arguments of different types. Not generic because dictionaries require non-generic types.
+///   </para>
+/// </remarks>
+public abstract class FilterArgument
+{
+    /// <summary>
+    ///   Helper function to get Number from NumberFilterArgument without cast.
+    /// </summary>
+    public float GetNumberValue()
+    {
+        return (this as NumberFilterArgument)?.Value ??
+            throw new InvalidOperationException("Can't get number value from a non-numeric filter argument!");
+    }
+
+    public string GetStringValue()
+    {
+        return (this as MultipleChoiceFilterArgument)?.Value ??
+            throw new InvalidOperationException("Can't get string value from a non-string filter argument!");
+    }
+
+    public class NumberFilterArgument : FilterArgument
+    {
+        public readonly float MinValue;
+        public readonly float MaxValue;
+
+        public float Value;
+
+        public NumberFilterArgument(float minValue, float maxValue, float defaultValue)
+        {
+            if (defaultValue < minValue || defaultValue > maxValue)
+                throw new ArgumentOutOfRangeException($"{defaultValue} is outside the range {minValue}-{maxValue}!");
+
+            MinValue = minValue;
+            MaxValue = maxValue;
+            Value = defaultValue;
+        }
+    }
+
+    public class MultipleChoiceFilterArgument : FilterArgument
+    {
+        public readonly List<string> Options;
+        public string Value;
+
+        public MultipleChoiceFilterArgument(List<string> options)
+        {
+            Options = options;
+            Value = Options.Count > 0 ? Options[0] : "--";
```

Comment:
`--` is not a good default value (I think)

Reply:
Should at least be translatable to other languages.

Reply:
Yes, I intend to change that a bit afterwards. I'm currently working on multiple filters at a time (and fixing snapshots with them)